addons/web/static/src/js/views/graph/graph_renderer.js

```diff
@@ -474,6 +474,9 @@ return AbstractRenderer.extend({
                         // don't use bind:  callback is called with 'index' as second parameter
                         // with value labels.indexOf(label)!
                         callback: function (label) {
+                            if(label && label[0] && typeof label[0]=='string' && label[0].length>15){
```

Comment:
Actually, I had forgotten but we can write instead
```suggestion
                            var fullText = self._relabelling(label);
                            return self._shortenLabel(fullText);
```
like in _getLegendOptions.
This is better and more general.